Install headers as part of default cmake install

nsmithtt opened the pull request.

AleksKnezevic commented:
All the files are in the install folder, but they don't point to one another, i.e. 
```
> ldd install/lib/libTTMLIR.so 
        linux-vdso.so.1 (0x00007ffc897cc000)
        libMLIR.so.19.0git => not found
        libLLVM.so.19.0git => not found
        _ttnn.so => not found
        libtt_metal.so => not found
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f65a3408000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f65a331f000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f65a32ff000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f65a30d6000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f65a3fee000)
```
if you link against the installed version of tt-metal libs, i.e.
```
diff --git a/lib/SharedLib/CMakeLists.txt b/lib/SharedLib/CMakeLists.txt
index e02dae0..2f6f1b0 100644
--- a/lib/SharedLib/CMakeLists.txt
+++ b/lib/SharedLib/CMakeLists.txt
@@ -47,7 +47,13 @@ target_link_libraries(TTMLIR PRIVATE
 
     flatbuffers
 )
-
-target_link_directories(TTMLIR PRIVATE ${TTMLIR_TOOLCHAIN_DIR}/lib)
+target_link_directories(TTMLIR PRIVATE 
+    ${TTMLIR_TOOLCHAIN_DIR}/lib
+    ${CMAKE_INSTALL_PREFIX}/lib
+)
+set_target_properties(TTMLIR
+    PROPERTIES
+    INSTALL_RPATH_USE_LINK_PATH TRUE
+)
 
 install(TARGETS TTMLIR DESTINATION ${CMAKE_INSTALL_LIBDIR})
```
then it's almost resolved, there are just a couple of tt-metal libs that are left pointing to the build dir...

AleksKnezevic commented:
Figured it out, had to add a few extra libs from ttmetal. I'll submit a pr once this lands.

nsmithtt commented:
@AleksKnezevic I also added `ORIGIN` to `libTTMLIR.so`'s rpath so now it should automatically find any libs inside the same directory.  I also added install component:
```
cmake --install build --component SharedLib 
```

Which installs only the minimal set of libs ie metal libs + `libTTMLIR.so`.
